enhancements/machine-config/pre-assigning-nodes-to-machine-config-pools-4-8.md

```diff
@@ -0,0 +1,269 @@
+---
+title: pre-assigning-nodes-to-machine-config-pools-4-8
+authors:
+  - "@beekhof"
+reviewers:
+  - "@slintes" 
+  - "@markmc"
+  - "@JoelSpeed"
+  - "@alexcrawford"
+  - "@kikisdeliveryservice"
+approvers:
+  - "@markmc"
+  - "@JoelSpeed"
+  - "@alexcrawford"
+  - "@kikisdeliveryservice"
+creation-date: 2021-03-31
+last-updated: 2021-03-31
+status: provisional
+see-also:
+  - "/openshift/enhancements/pull/717"
+replaces:
+  - 
+superseded-by:
+  - 
+---
+
+# Short-term approach for Pre-assigning Nodes to Machine Config Pools
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+In telco bare-metal environments, there is a need to have Nodes be provisioned
+with a known Machine Config Pool.
+
+The traditional flow is for Nodes to come up as workers and be moved to the
+desired pool after provisioning, however this consumes a significant portion of
+the maintenance window in bare-metal environments.
+
+## Motivation
+
+In telco environments, there is a fixed window (typically 4 hours, half of which
+is reserved for rollback) during which new hardware needs to be installed and
+provisioned.
+
+In the case of remote worker nodes, they will typically need to be a part of a
+specific (non-default) MachineConfigPool, and are provisioned with the correct
+pool based on the Ingnition file contents.
+
+However, the MCO/MCD normally manages MCP assignment and does so based on Node
+Labels.  This creates a race to add the necessary labels before the MCO/MCD
```

Comment:
maybe add a sentence in between, that adding node labels during installation using Ignition is not possible

Reply:
Technically it is possible (see alternatives) but has other undesirable properties :)